core/commands/block.go

```diff
@@ -157,27 +159,32 @@ It reads from stdin, and <key> is a base58 encoded multihash.
 			return
 		}
 
+		mhtype, _ := req.Options["mhtype"].(string)
+		mhtval, ok := mh.Names[mhtype]
+		if !ok {
+			err := fmt.Errorf("unrecognized multihash function: %s", mhtype)
+			res.SetError(err, cmdkit.ErrNormal)
+			return
+		}
+
 		var pref cid.Prefix
 		pref.Version = 1
 
-		format, _ := req.Options["format"].(string)
+		format, userFormat := req.Options["format"].(string)
```

Comment:
Unfortunately, `userFormat` will always be true there is a default value. The only way I know to fix this, is to simply remove the default.